src/NServiceBus.Transport.Msmq/MsmqConfigurationExtensions.cs

```diff
@@ -139,5 +139,86 @@ internal static bool GetShouldExecuteInstaller(this SettingsHolder settings)
             }
             return true;
         }
+
+        /// <summary>
+        /// Disables the storing of undeliverable messages in the dead letter queue
+        /// </summary>
+        /// <param name="config"></param>
+        public static void DoNotUseDeadLetterQueue(this TransportExtensions<MsmqTransport> config)
+        {
+            Guard.AgainstNull(nameof(config), config);
+            config.GetSettings().Set("UseDeadLetterQueue", false);
+        }
+
+        internal static bool GetUseDeadLetterQueue(this SettingsHolder settings)
+        {
+            return settings.GetOrDefault<bool>("UseDeadLetterQueue");
+        }
+
+        /// <summary>
+        /// Does not cache connections. 
+        /// </summary>
+        /// <param name="config"></param>
+        public static void DoNotCacheConnections(this TransportExtensions<MsmqTransport> config)
+        {
+            Guard.AgainstNull(nameof(config), config);
+            config.GetSettings().Set("UseConnectionCache", false);
+        }
+
+        internal static bool GetUseConnectionCache(this SettingsHolder settings)
+        {
+            return settings.GetOrDefault<bool>("UseConnectionCache");
+        }
+
+        /// <summary>
+        /// This setting should be used with caution. As the queues are not transactional, any message that has
+        /// an exception during processing will not be rolled back to the queue. Therefore this setting must only
+        /// be used where loss of messages is an acceptable scenario.  
+        /// </summary>
+        /// <param name="config"></param>
+        public static void DoNotUseTransactionalQueues(this TransportExtensions<MsmqTransport> config)
+        {
+            Guard.AgainstNull(nameof(config), config);
+            config.GetSettings().Set("UseTransactionalQueues", false);
+        }
+
+        internal static bool GetUseTransactionalQueues(this SettingsHolder settings)
+        {
+            return settings.GetOrDefault<bool>("UseTransactionalQueues");
+        }
+
+        /// <summary>
+        /// Enables the use of journaling messages. Stores a copy of every message received in the journal queue. 
+        /// Should be used ONLY when debugging as it can 
+        /// potentially use up the disk storage quota based on the message volume.
```

Comment:
How about "MSMQ disk storage quota"

Reply:
It's actually the MSMQ Journal quota. It's separate from the regular MSMQ quota. 

Added the info

Reply:
Oh really? Had no idea. Thanks